RHEL6_7/system/initscripts/ifcfg/check

```diff
@@ -51,99 +51,114 @@ I case 3, to be completely correct you could check that user is not using rhel7
+            preexec_fn=None,
+            close_fds=True,
+            shell=False,
+            cwd=None,
+            env=None,
+            universal_newlines=False,
+            startupinfo=None,
+            creationflags=0).wait()
+        except:
+            output.seek(0)
+            err(output)
+            raise
+        output.seek(0)
+        if return_code < 0:#error
+            log_error(output.read())
+            raise err_msg
+        if return_code == 1:#undefined variable
+            return False
+        return output.read().strip()
+
+    def ls_scripts():
+        for f in [i for i in os.listdir(ifcfg_path) if i.startswith(ifcfg_prefix)]:
+            yield f
+    
+    def is_kernel(name):
+        return re.match("eth[0-9]+", name)
+
+    def is_udev(name):
+        if name[0:2] in ("en", "sl", "wl", "ww"):
+            if name[2] in ("b", "c", "o", "s", "x", "P", "p"):
+                return True
+        return False
+
+    def fix_ifnames():
+        script_result = subprocess.Popen("./preupg_network.sh", shell=True)
+        script_result.communicate()
+        exit_code = script_result.returncode
+        if exit_code == 1:
+            log_extreme_risk("The network interface names on the source system are in inconsistent state. Upgrade is not possible")
+        elif exit_code == 2:
+            log_extreme_risk("MACADDR directive found in the ifcfg files. Comment it out and rerun preupgrade-assistant Upgrade is not possible")
+
+    ethx_with_addr_count = 0
+    ethx_without_addr_count = 0
+    ifnames_to_fix = 0
+    warning = False
+    for script in ls_scripts():
+        full_path = ifcfg_path + script
+        addr = get_variable(full_path, "HWADDR")
+        name = get_variable(full_path, "NAME" if devel_time else "DEVICE")
+        log_debug("checking " + script + ", name: " + str(name) + " hwaddr: " + str(addr))
+        short_name = script[len(ifcfg_prefix):]
+        if short_name == "lo": #loopback
+            continue
+        if not name:
+            warning = True
+            if not addr:
+                log_slight_risk(full_path + " does not have DEVICE nor HWADDR set, what kind of device is it?")
+            else:
+                log_slight_risk(full_path + " does not have DEVICE set, its name can change after the upgrade.")
+        else:
+            if is_kernel(name):
+                if addr:
+                    ethx_with_addr_count +=1
+                else:
+                    ethx_without_addr_count +=1
+                    log_slight_risk(full_path + " is old style ethX name without HWADDR, its name can change after the upgrade.")
+                    warning = True
+            elif is_udev(name):
+                log_slight_risk(full_path + " variable DEVICE is similar to udev predictable network naming scheme. It might cause conflicts.")
+                warning = True
+
+        if 1 in (ethx_with_addr_count, ethx_without_addr_count) > 1:
```

Comment:
What do you want this condition to do? I would there's a 99% chance this construct does not do what you want. Or at least it is very confusing.

I checked the precedence of operators in Python and the `in` is on the same level of precedence as the `>` operator. Then it the operators are resolved from left to right. So:
`1 in (ethx_with_addr_count, ethx_without_addr_count)` returns `True` (which in Python is the same as `1`) if either of the two variables holds `1`, and then you're comparing either `True` or `False` to be bigger then `1` which is always `False`.

Reply:
You just want to know if either x or y has value bigger than 1, so I would write:
`if ethx_with_addr_count > 1 or ethx_without_addr_count > 1:`

Reply:
Thanks for the review on this. I have corrected the statement in a new build